test/byContract/registries/registries.test.ts

```diff
@@ -0,0 +1,330 @@
+// SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { BytesLike } from '@ethersproject/bytes';
+import { AddressZero } from '@ethersproject/constants';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import hre, { deployments, ethers, expect } from 'hardhat';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import {
+    AppRegistry,
+    AppRegistry__factory,
+    DatasetRegistry,
+    DatasetRegistry__factory,
+    IexecInterfaceNative,
+    IexecInterfaceNative__factory,
+    WorkerpoolRegistry,
+    WorkerpoolRegistry__factory,
+} from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+const constants = require('../../../utils/constants');
+const randomAddress = () => ethers.Wallet.createRandom().address;
+
+describe('Registries', () => {
+    let proxyAddress: string;
+    let [iexecPoco, iexecPocoAsAdmin]: IexecInterfaceNative[] = [];
+    let [iexecAdmin, appProvider, datasetProvider, scheduler, anyone]: SignerWithAddress[] = [];
+
+    let ensRegistryAddress: string;
+    let appRegistry: AppRegistry;
+    let datasetRegistry: DatasetRegistry;
+    let workerpoolRegistry: WorkerpoolRegistry;
+
+    beforeEach(async () => {
+        proxyAddress = await loadHardhatFixtureDeployment();
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        ({ iexecAdmin, appProvider, datasetProvider, scheduler, anyone } =
+            await getIexecAccounts());
+        ensRegistryAddress = (await deployments.get('ENSRegistry')).address;
+        const appRegistryAddress = (await deployments.get('AppRegistry')).address;
+        appRegistry = AppRegistry__factory.connect(appRegistryAddress, iexecAdmin);
+        const datasetRegistryAddress = (await deployments.get('DatasetRegistry')).address;
+        datasetRegistry = DatasetRegistry__factory.connect(datasetRegistryAddress, iexecAdmin);
+        const workerpoolRegistryAddress = (await deployments.get('WorkerpoolRegistry')).address;
+        workerpoolRegistry = WorkerpoolRegistry__factory.connect(
+            workerpoolRegistryAddress,
+            iexecAdmin,
+        );
+
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+    }
+
+    describe('Registry', () => {
```

Comment:
Remove this describe then 1 describe per function?

Reply:
Done [here](https://github.com/iExecBlockchainComputing/PoCo/pull/122/commits/812e972a8d71789673e739a09b8f77f66648a42c) tell me if that's okay for you